File: innerlibs1/htmlparser/CSSSelectors/PseudoClassSelectors/LastChildPseudoClass.vb

Imports HtmlParser
Imports System.Collections.Generic
Imports System.Linq
Imports System.Text
Imports System.Threading.Tasks

Namespace HtmlParser.PseudoClassSelectors
    <PseudoClassName("last-child")>
    Friend Class LastChildPseudoClass
        Inherits PseudoClass
        Protected Overrides Function CheckNode(node As HtmlElement, parameter As String) As Boolean
            Return node.Parent.Nodes().Count = node.Index - 1
        End Function
    End Class
End Namespace

'=======================================================
'Service provided by Telerik (www.telerik.com)
'Conversion powered by NRefactory.
'Twitter: @telerik
'Facebook: facebook.com/telerik
'=======================================================

File: InnerLibsCommon/HtmlParser/PseudoClassSelectors/LastChildPseudoClass.cs
using System.Linq;

namespace Extensions.Web.PseudoClassSelectors
{
    [PseudoClassName("last-child")]
    internal class LastChildPseudoClass : PseudoClass
    {
        protected override bool CheckNode(HtmlNode node, string parameter) => node.ParentNode.GetChildElements().Last() == node;
    }
}